subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -4062,7 +4062,9 @@ static void socket_receive_loop(void)
 	while (1) {
 		/* wait for sockets */
 		if (sock_nfds < 1) {
-			k_sleep(lwm2m_engine_service());
+			int delay = lwm2m_engine_service();
+
+			k_sleep(K_MSEC(delay));
```

Comment:
This certainly works. There is also `k_msleep(msec);` function that could be used here too. I leave it you to decide if you would like to use that or not.

Reply:
I'll swtich to `k_msleep`.